src/tracers.jl

```diff
@@ -13,19 +11,24 @@ empty(tracer::T) where {T<:AbstractTracer} = empty(T)
     ConnectivityTracer{S}(indexset) <: Number
 
 Number type keeping track of input indices of previous computations.
-The provided index set type `S` has to be an `AbstractSet{<:Integer}`, e.g. `BitSet` or `Set{UInt64}`.
+The provided index set type `S` has to satisfy the following conditions:
+
+- it is an iterable with `<:Integer` element type
+- it implements methods `union` and `push!`
```

Comment:
> it implements methods `union` and `push!`

I just checked: `union`, `union!` and `push!` are required

Reply:
why `union!`? when do you ever mutate a `Tracer` (which is essentially a number) in-place?

Reply:
For the `HessianTracer`, I take a `deepcopy` of one of the dicts and mutate it in place. It doesn't have to be this way, but it currently is.

Reply:
As mentioned in #34, the `HessianTracer` leaves room for optimization.

Reply:
How about this custom dict merging?

```julia
julia> d1 = Dict(:a => Set([1, 2, 3]), :b => Set([3, 4, 5]))
Dict{Symbol, Set{Int64}} with 2 entries:
  :a => Set([2, 3, 1])
  :b => Set([5, 4, 3])

julia> d2 = Dict(:b => Set([5, 6, 7]), :c => Set([8, 9, 10]))
Dict{Symbol, Set{Int64}} with 2 entries:
  :b => Set([5, 6, 7])
  :c => Set([10, 9, 8])

julia> mergewith(union, d1, d2)
Dict{Symbol, Set{Int64}} with 3 entries:
  :a => Set([2, 3, 1])
  :b => Set([5, 4, 6, 7, 3])
  :c => Set([10, 9, 8])
```

Reply:
See #35 for this discussion, I have added `union!` in the meantime

Reply:
Commented on in #35. Your proposed `mergewith` already exists and is called `additive_merge`. `union!` is used as inside the current implementation of `additive_merge`.